### PR TITLE
Immanuel

### PR DESCRIPTION
**Pull request by ImmanuelCurrah**

### CODE REVIEW

**Comment by ImmanuelCurrah** [comment]
Looks fucking good! :D just cleaned it up a bit. But honestly really good find with the useMemo and useContext :) I forgot you were a genius haha

**Comment by mike701** [comment]
Lol thanks bro I have one thing I wanna change here because I noticed it after I saw your pr. I will make the suggestion just let me know what you think. I believe this will save us some hassle.

File: TypeScript/src/App.tsx [hunk 1]
```diff
@@ -1,46 +1,52 @@
...
 
-function createPageContext(pageNum, setPageNum) {
-  // const [pageNum, setPageNum] = useState<number | null | undefined>(1);
+function createPageContext(
+  pageNum: number,
+  setPageNum: Dispatch<SetStateAction<number>>
+) {
   if (localStorage.getItem("pageId")) {
-    setPageNum(localStorage.getItem("pageId"));
+    setPageNum(+localStorage.getItem("pageId"));
```
Comment (mike701) [inline TypeScript/src/App.tsx]: I think we should check for a diff between the getItem and the pageNum we are setting. Then we can simply setItem if there is a diff and thus less hassle.
`if (pageNum !== localStorage.getItem("pageId")) {
      setPageNum(pageNum);
      localStorage.setItem("pageId", pageNum);
    }`

Comment (ImmanuelCurrah) [reply TypeScript/src/App.tsx]: Yeah cleaner for sure

**Review by mike701 — CHANGES_REQUESTED** [review]
Beyond the comments I left about adding that check and the change to page num I think pr looks good

File: TypeScript/src/App.tsx (hunk 1 above)
Comment (ImmanuelCurrah) [reply TypeScript/src/App.tsx]: changed

**Review by mike701 — APPROVED** [review]

**Comment by ImmanuelCurrah** [comment]
Yeah we are getting there lol. I think this was a good thing to put some thought into. Will make it much less of a hassle when the app grows

**Pull request merged**